pyvista/core/filters.py

```diff
@@ -1953,6 +1953,10 @@ def streamlines(dataset, vectors=None, source_center=None,
             The coordinates of a start and end point for a line source. This
             will override the sphere point source.
 
+        source : pyvista.DataSet <- TODO this isn't correct, what is the right type?
```

Comment:
By my testing, not all DataSet work, see more comments below.  I'm not really sure how to handle this.